neptune/internal/utils/git_info.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (c) 2019, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import sys
+import subprocess
+
+
+class GitInfo(object):
+    def __init__(self, commit_id, message, author_name, author_email, commit_date, repository_dirty):
+        self.commit_id = commit_id
+        self.message = message
+        self.author_name = author_name
+        self.author_email = author_email
+        self.commit_date = commit_date
+        self.repository_dirty = repository_dirty
+
+
+def get_git_info():
+    try:
+        import git
+
+        def get_git_version():
+            try:
+                with open(os.devnull, 'w') as devnull:
+                    return subprocess.check_output(['git', '--version'], stderr=devnull).decode("utf-8").strip()
+            except:  # pylint: disable=bare-except
+                return None
+
+        if not get_git_version():
+            return None
+
+        if hasattr(sys, 'getwindowsversion') and r'GIT_PYTHON_GIT_EXECUTABLE' not in os.environ:
+            os.environ[r'GIT_PYTHON_GIT_EXECUTABLE'] = os.popen("where git").read().strip()
+
+        repository_path = os.getcwd()
```

Comment:
Is this always the case? the CWD doesn't have to have anything to do with where the repo is.

Reply:
What better option do you suggest? please keep in mind that this is an internal implementation for default git info - user is not supposed to call it manually